### DISCUSSION TITLE
Strange problem when page.cropbox changed

### DISCUSSION THREAD

**Discussion by mlove4u** [pymupdf/PyMuPDF#1269] [Looking for help] [answered]
OS: macOS 10.15.7
PyMuPDF: 1.18.17
Python: 3.7.4

### My use case:
1. change the value of page.cropbox to trimbox
2. render a image from pdf page.  Rendering area -->page.cropbox(trimbox)

### Problem:
**page.get_pixmap(clip=page.mediabox)** and **page.get_pixmap(clip=page.cropbox)** does not crop the right area as expected.
Please take a look at the attached png file. Sorry for my poor English. Thanks.
Here is my code:
```
import fitz
from PIL import Image


def show_pm(pm, file_name):
    pil = Image.frombuffer('RGB', (pm.width, pm.height), pm.samples)
    pil.save(file_name)
    pil.show()


doc = fitz.open("trimbox.pdf")
page = doc[0]
#mat1 = page.transformation_matrix
#
print("old page.cropbox: ", page.cropbox)
# >> old page.cropbox:  Rect(0.0, 0.0, 400.0, 200.0)
pm = page.get_pixmap()  # clip(default)=page.cropbox
show_pm(pm, "0_page.cropbox.png")
#
pm = page.get_pixmap(clip=page.mediabox)
show_pm(pm, "0_page.mediabox.png")
#
#
doc.xref_set_key(page.xref, "CropBox", '[50 50 350 150]')  # change cropbox
print("new page.cropbox: ", page.cropbox)
# >> new page.cropbox:  Rect(50.0, 50.0, 350.0, 150.0)
#mat2 = page.transformation_matrix
pm = page.get_pixmap(clip=page.mediabox)  # page.mediabox.irect
show_pm(pm, "1_page.mediabox.png")
#
pm = page.get_pixmap(clip=page.cropbox)
show_pm(pm, "1_page.cropbox.png")
#
pm = page.get_pixmap()
show_pm(pm, "1_.png")

```
![pngs](https://user-images.githubusercontent.com/18019314/133719595-17415e91-7ca3-4a4e-b898-7a541a485e9f.png)
[trimbox.pdf](https://github.com/pymupdf/PyMuPDF/files/7182385/trimbox.pdf)

**Comment by JorjMcKie** [comment]
I will have a look.
Meanwhile I will convert htis to a discussion.

**Comment by JorjMcKie** [comment]
It is early in the morning here, but to me everything works as expected - nothing is strange.
If you specify a clip parameter, then its intersection with the page rectangle will be taken. In so far, the formulation in the documentation is a bit misleading: I should have written "page rectangle".

**Reply by mlove4u** [reply]
Thank you for your reply.
What do you mean by "page rectangle"? Does it means **Page.rect**?
I noticed that the value of Page.rect changed from (0.0, 0.0, 400.0, 200.0) to **(0.0, 0.0, 300.0, 100.0)** (-->A_0-0.png) after I replaced the value of Page.cropbox.
But, what is the area of new Page.rect? It seems to be the pink area of  **B_50-50.png** but not A_0-0.png. 
**B_50-50.png**
<img width="679" alt="B_50-50" src="https://user-images.githubusercontent.com/18019314/133788780-ff4fa132-fb6a-4b6e-9ea2-6d69e1231857.png">
**A_0-0.png**
<img width="659" alt="A_0-0" src="https://user-images.githubusercontent.com/18019314/133788830-6aa6cbf4-9630-4ecd-8d5f-f90b0628ea73.png">

Does the origin(0,0) of Page.rect changes depending on coordinate of  Page.cropbox ?
Thanks.

**Comment by JorjMcKie** [comment, marked as answer]
Please consult the documentation!
Potntially except for page rotation and the top-left value, page.rect equals CropBox. The page may be rotated, but the CropBox never is. Width and height of the unrotated page always equal those of cropbox.
The top-left of page.rect always is (0,0).
So if you change the cropbox, you also change page.rect.

**Reply by mlove4u** [reply]
I got. Thank you very much.

**Comment by JorjMcKie** [comment]
CropBox is also the default for TrimBox, ArtBox and BleedBox. If CropBox is missing then MediaBox is used. See Adobe PDF guide.
